test/unit/test_utils.py

```diff
@@ -238,6 +238,35 @@ def test_get_tokens(line, expected_tokens):
     tokens = list(utils.get_tokens(line))
     assert tokens == expected_tokens
 
+def test_combine_means():
+    error_tolerance = 1e-05
+    n_lists = np.random.randint(1, 10)
```

Comment:
Thanks for adding tests! I think for tests we should avoid randomness to make results deterministic. We could either fix a seed or simply specify a list of fixed data. I'd prefer the latter as it's more explicit and wouldn't fail if the underlying random number generator would change in the future. 

Given that the inputs can also be `None`, we could use `@pytest.mark.parametrize` (see above) to specify some lists where some entries are None to test this works as expected..